### PR TITLE
Add m3u playlist download

### PR DESCRIPTION
**Pull request by Tereius**

### CODE REVIEW

File: src/routes/station.ts [hunk 1]
```diff
@@ -24,4 +25,15 @@ router.get('/download', (req: Request, res: Response) => {
...
 });
 
+router.get('/downloadm3u', (req: Request, res: Response) => {
+    res.setHeader('Content-Type', 'application/x-mpegURL');
+    res.setHeader('Content-Disposition', 'attachment; filename="IPTV-ReStream.m3u"');
+    const protocol = config['xspf_protocol'] || req.protocol;
+    const host = config.xspf_host || req.get('X-Forwarded-Host') || req.get('Host');
+    const pathPrefix = config.xspf_pathPrefix;
+    res.write(m3uProvider.generateM3U(`${protocol}://${host}${pathPrefix}`, req.query.simple &&
+        "TRUE" === req.query.simple.toString().toUpperCase() ? true : false));
```
Comment (n-thumann) [inline src/routes/station.ts:35]:
```suggestion
    res.write(m3uProvider.generateM3U(`${protocol}://${host}${pathPrefix}`));
```

File: src/providers/m3u.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+        let m3uString = '#EXTM3U' + '\n';
+        let chNum = 1;
+
+        const stations = stationProvider.stations;
+        for (const station of stations) {
+            m3uString += '#EXTINF:-1';
+            if (!simple) {
+                m3uString += ` tvg-chno="${chNum}" tvg-name="${station.title}" tvg-logo="${station.image}" group-title="IPTV"`;
+            }
```
Comment (n-thumann) [inline src/providers/m3u.ts:14]:
```suggestion
            m3uString += ` tvg-chno="${chNum}" tvg-name="${station.title}" tvg-logo="${station.image}" group-title="IPTV"`;
```

File: src/providers/m3u.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import stationProvider from './station';
+
+class M3UProvider {
+    public generateM3U(hostname: string, simple: boolean) {
```
Comment (n-thumann) [inline src/providers/m3u.ts:4]:
```suggestion
    public generateM3U(hostname: string) {
```

**Review by n-thumann — CHANGES_REQUESTED** [review]
Thank you very much for your contribution! That look´s like a useful feature! 👏 
I'd prefer removing the `simple` parameter, because it isn't accessible from the UI and I think it isn't much of a benefit to not include the channel number, title and image in the M3U. Of course it takes up three times more space, but that should be acceptable as we're talking kilobytes here 😊

**Comment by Tereius** [comment]
I agree. Thank you for merging.

**Comment by n-thumann** [comment]
Merged and pushed a new release ✌️

**Pull request merged**